applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js

```diff
@@ -144,22 +169,55 @@ function collectWlanAPInfos(compactconnectioninfo_table_entries, wlanAPInfos) {
 	}
 };
 
+var callNetworkRrdnsLookup = rpc.declare({
+	object: 'network.rrdns',
+	method: 'lookup',
+	params: [ 'addrs', 'timeout', 'limit' ],
+	expect: { '': {} }
+});
+
+
 function collectRemoteHosts (remotehosttableentries,Remotehosts) {
+	var ipAddrs = [];
 	for (var IPaddr in Remotehosts) {
-			remotehosttableentries.push([IPaddr, Remotehosts[IPaddr]['id']]);
-		}
+		if (typeof dns_cache[IPaddr] === 'undefined') ipAddrs.push(IPaddr);
+	}
+
+	L.resolveDefault(callNetworkRrdnsLookup(ipAddrs, 1000, 1000), {}).then(function(replies) {
+				for (var index in ipAddrs) {
+					var address = ipAddrs[index];
+					if (!address)
+						continue;
+					if (replies[address]) {
+						dns_cache[address] = replies[address];
+						continue;
+					} else {
+						for (var IPaddr in Remotehosts)
+							for (var mac in Hosts) {
```

Comment:
I think this should be an `of` loop

Reply:
actually no, i think "in" is the correct usage here.  Agree?
`root@OpenWrt6:~# ubus call usteer "remote_hosts"
{
        "192.168.64.2": {
                "id": -454639439
        },
        "192.168.64.1": {
                "id": -812651663
        }
}
root@OpenWrt6:~# ubus call luci-rpc "getHostHints"
{
        "00:11:22:33:44:55": {
                "ipaddrs": [
                        "192.168.64.65"
                ],
                "ip6addrs": [

                ],
                "name": "xyz"
        },

'

Reply:
I think `of`. you want the actual IP, and not a loop counting integer, correct?